src/api/coderApi.ts

```diff
@@ -74,38 +74,63 @@ export class CoderApi extends Api {
 		output: Logger,
 	): CoderApi {
 		const client = new CoderApi(output);
-		client.setHost(baseUrl);
-		if (token) {
-			client.setSessionToken(token);
-		}
+		client.setCredentials(baseUrl, token);
 
 		setupInterceptors(client, output);
 		return client;
 	}
 
-	setSessionToken = (token: string): void => {
-		const defaultHeaders = this.getAxiosInstance().defaults.headers.common;
-		const currentToken = defaultHeaders[coderSessionTokenHeader];
-		defaultHeaders[coderSessionTokenHeader] = token;
+	/**
+	 * Set both host and token together. Useful for login/logout/switch to
+	 * avoid triggering multiple reconnection events.
+	 */
+	setCredentials = (
+		host: string | undefined,
+		token: string | undefined,
+	): void => {
+		const defaults = this.getAxiosInstance().defaults;
+		const currentHost = defaults.baseURL;
+		const currentToken = defaults.headers.common[coderSessionTokenHeader];
 
-		if (currentToken !== token) {
+		defaults.baseURL = host;
+		defaults.headers.common[coderSessionTokenHeader] = token;
+
+		const hostChanged = currentHost !== host;
+		const tokenChanged = currentToken !== token;
```

Comment:
Do we need to handle the case of comparing an empty string to undefined?  That could trigger a reconnect or `suspend` but maybe going from falsey to falsey should be considered as no change.

Reply:
Could you expand on that use case?  If we are `undefined` then we call `socket.suspend()`, if we then go to `""` we call `socket.suspend()` a second time which I assume is a no-op?

Reply:
For the host, that is.

For the token, this would add a blank token to the headers but this has no effect in the mTLS case I think?  Even in the non-mTLS case the outcome would be the same (the server makes no distinction between empty or missing, both would be a 401).